src/gui/transferlistsortmodel.cpp

```diff
@@ -109,15 +110,19 @@ bool TransferListSortModel::lessThan_impl(const QModelIndex &left, const QModelI
     };
 
     const int sortColumn = left.column();
+    // make sure not to call invokeLessThanForColumn when subSort == TransferListModel::TR_QUEUE_POSITION || subSort == TransferListModel::TR_NAME
+    // that may cause infinite recursion
+    const int subSortColumn = BitTorrent::Session::instance()->isQueueingSystemEnabled()
+                              ? TransferListModel::TR_QUEUE_POSITION : TransferListModel::TR_NAME;
     const QVariant leftValue = left.data(TransferListModel::UnderlyingDataRole);
     const QVariant rightValue = right.data(TransferListModel::UnderlyingDataRole);
 
     switch (sortColumn) {
     case TransferListModel::TR_CATEGORY:
     case TransferListModel::TR_TAGS:
     case TransferListModel::TR_NAME:
-        if (!leftValue.isValid() || !rightValue.isValid() || (leftValue == rightValue))
-            return invokeLessThanForColumn(TransferListModel::TR_QUEUE_POSITION);
+        if ((!leftValue.isValid() || !rightValue.isValid() || (leftValue == rightValue)) && subSortColumn != TransferListModel::TR_NAME)
+            return invokeLessThanForColumn(subSortColumn);
```

Comment:
Looks like you broke all the remaining fallbacks. I think you should keep this branch unchanged.